### ISSUE TITLE
Mob resistance doesn't check against proper cap

### ISSUE THREAD

**Issue by Shadownflame** [Project-Epoch/bugtracker#8344]
Mob spell resistances always seem to calculate resist % chances from the mob's own level and not enemy level.
So a Stormwind guard (ID 1756) with 50 resistance against Frost magic is giving the intended 10-20% partial resists for a level 60 player because the Guard himself is lvl 60 and the 50 resistance calculates against a cap of 300:
![WoWScrnShot_072624_025643](https://github.com/user-attachments/assets/d4559ded-9d98-43ad-96ec-090a2338a86f)
But a lvl 15 mob with 50 frost resistance is returning 40-60% partial resists for a lvl 60 enemy because it seems to calculate with the mob's own level, 50 resist against 75 cap:
![WoWScrnShot_072624_032155](https://github.com/user-attachments/assets/eacdeacb-380c-4a5f-9c33-f5c0c392900f)

Resistance value should always be checked against a cap calculated from the attacker's level, in this case a lvl 60 attacker should always be seeing the values that belong to 50/300 resistance, not 50/75 or anything else.

**Comment by ghost** [comment]
If anyone has more information about this I would appreciate it. I looked at Vmangos, CMangos-TBC, and AC. They all check the victim's cap, not the attacker's cap.

I think it's possible that it is the highest of the two numbers, checking attacker's cap only if that cap is higher than the victim's cap. But I find it hard to believe all of these cores have such a basic mechanic wrong

**Comment by Shadownflame** [comment]
TL;DR yes it has been most likely broken on stock pservers since forever.

Basically every source from retail ever speaks of 315 resistance cap against bosses (resistance calculators are set up this way too), which is due to boss level being 63 and calculated as the cap when the boss is attacking you, i.e. when tanking twin emps. This is basic knowledge, not a matter of debate, every retail source firmly uses the attacker's (boss') level as cap, I can confirm this.
However, some private server players who had done insider testing on Elysium confirm a cap of 300 instead (the original discussion is deleted but the conclusion was copied over to reddit here https://www.reddit.com/r/lightshope/comments/7ver73/pve_resistances/?utm_source=reddit-android ), which aligns with my personal experience on every private server ever, that getting any resistance above 300 was absolutely wasted and translated to no further damage mitigation (I didn't do detailed numerical tests to prove this, it's just in the category of "everyone feels and knows")

Everyone who ever played on vanilla pservers also experienced the issue of low lvl mobs resisting half your damage even if you outlevel them, as shown on the screenshot I posted from Epoch internal above. It has always been a thing, and always impacted leveling experience from casters (the power of mob resistances not weaning off as you outlevel the enemies in the world).
Here is a detailed post discussing this and pserver resistance problems in general: https://www.reddit.com/r/wowservers/comments/59tvha/cmangos_presents_first_fully_blizzlike_magic/

And here is another discussion on this matter: https://www.reddit.com/r/lightshope/comments/at24r5/is_the_resistance_cap_300_or_315/ it seems that light's hope server eventually noticed this problem and fixed it (I didn't play on light's hope personally), and these redditers provided some other useful links to prove this:

https://github.com/lh-server/core/blob/f283ef4dcab9d3cdf1b517bf9d45fa3194247bea/src/game/Objects/Unit.cpp#L3119
(copy pasting laipro's comment from reddit here outlining the important part):
damage reduction is capped between 0% and 75% (3121-3124), here it is a float number, capped between 0,0 and 0,75.
the line 3119 is basicly:
damage_reduction_modifier = victum_resistance * ( 0,15 / (attacker_level) )
to achieve max damage reduction when attacked by lvl63 you need 315 resistance.

Why has this remained broken on the base cores for so long?
The difference between the boss cap calculating with 300 or 315 is about 2% overall damage mitigation, which doesn't make a dramatic difference for the typical overgeared private server raiding party, never quite poked out anyone's eyes, so servers of lesser quality never cared to investigate or fix this. It's similar to the 3.3.5 expertise cap which has been 22 instead of 26 for a good ten years before some servers bothered fixing it.
